docs/introduction-to-smart-contracts.rst

```diff
@@ -502,10 +502,10 @@ depth limit of a little less than 1000 in practice.
 Delegatecall / Callcode and Libraries
 =====================================
 
-There exists a special variant of a message call, named **delegatecall**
-which is identical to a message call apart from the fact that
-the code at the target address is executed in the context of the calling
-contract and ``msg.sender`` and ``msg.value`` do not change their values.
+There exists a special variant of a message call, named **delegatecall** which is identical to a message call
+apart from the fact that the code at the target address is executed in the context of the calling contract.
+In practice this means ``msg.sender`` and ``msg.value`` retain the values that were passed to the delegator,
```

Comment:
It sounds a bit off to talk about ``msg.sender`` and ``msg.value`` being "passed to the delegator".
I'm not sure splitting this sentence really has that much merit in general.